mavros/include/mavros/setpoint_mixin.h

```diff
@@ -193,7 +199,48 @@ class TF2ListenerMixin {
 						tf_transform_cb(transform);
 					}
 					catch (tf2::LookupException &ex) {
-						ROS_ERROR_NAMED("tf2_buffer", "%s: %s", tf_thd_name.c_str(), ex.what());
+						ROS_ERROR_NAMED("tf2_buffer", "%s: %s", tf_thd_name_1.c_str(), ex.what());
+					}
+				}
+				rate.sleep();
+			}
+		});
+	}
+
+	/**
+	 * @brief start tf listener syncronized with another topic
+	 *
+	 * @param _thd_name  listener thread name
+	 * @param cbp        plugin callback function
+	 */
+	template <class T>
+	void tf2_start(const char *_thd_name, void (D::*cbp)(const geometry_msgs::TransformStamped &, const typename T::ConstPtr &))
+	{
+		tf_thd_name_2 = _thd_name;
+
+		tf_thread_2 = std::thread([this, cbp]() {
+			mavconn::utils::set_this_thread_name("%s", tf_thd_name_2.c_str());
+
+			mavros::UAS *m_uas_ = static_cast<D *>(this)->m_uas;
+			ros::NodeHandle &_sp_nh = static_cast<D *>(this)->sp_nh;
+			std::string &_frame_id = static_cast<D *>(this)->tf_frame_id;
+			std::string &_child_frame_id = static_cast<D *>(this)->tf_child_frame_id;
+
+			message_filters::Subscriber<T> _topic_sub;
+
+			ros::Rate rate(static_cast<D *>(this)->tf_rate);
+			while (ros::ok()) {
+				// Wait up to 3s for transform
+				if (m_uas_->tf2_buffer.canTransform(_frame_id, _child_frame_id, ros::Time(0), ros::Duration(3.0))) {
+					try {
+						auto transform = m_uas_->tf2_buffer.lookupTransform(
+								_frame_id, _child_frame_id, ros::Time(0), ros::Duration(3.0));
+
+						tf2_ros::MessageFilter<T> tf2_filter(_topic_sub, m_uas_->tf2_buffer, _frame_id, 10, _sp_nh);
+						tf2_filter.registerCallback(boost::bind(cbp, static_cast<D *>(this), transform, _1));
```

Comment:
I had `_top_sub` initialized before in a proper way. I don't know why I changed it. I would roll back and initialize it on plugin and use it here on the template by applying `static_cast<D *>(this)->`. Sorry for that